Add node name docs (#4552)
ydb-platform/ydb: ydb/docs/en/core/deploy/configuration/config.md

`mirror-3-dc` | Rack | Data center | 3 | 4 in each data center | Can stand a failure of a data center and 1 rack in one of the two other data centers
`mirror-3-dc` | Server | Data center | 3 | Doesn't matter | Can stand a failure of a data center and 1 server in one of the two other data centers

## Node Broker Configuration {#node-broker-config}

> Node Broker is a system tablet that is responsible for registering dynamic nodes in the {{ ydb-short-name }} cluster.
> 
> Node Broker can automatically generate names for dynamic nodes, which identify the node within the tenant and are displayed in metrics instead of the host name. If a dynamic node has been shut down, after a timeout its name can be taken by a new dynamic node. This allows node names to be reused when host names are changed.
> 
> To enable automatic node name generation, you need to add the following to the configuration:

Reviewer: ```suggestion
Node Broker is a system tablet that registers dynamic nodes in the {{ ydb-short-name }} cluster.

Node Broker can automatically generate artificial names for dynamic nodes. These names identify the node within the tenant and are displayed in metrics instead of the hostname. If a dynamic node has been shut down after a timeout, its name can be taken by a new dynamic node. This fact allows node names to be reused when host names are changed. Without this feature, the number of known dynamic node names in a cluster can grow infinitely if new dynamic nodes always get unique hostnames, which is common in cloud environments. With this feature, this number has an upper bound roughly equal to the peak number of dynamic nodes working simultaneously.

To enable the automatic generation of artificial dynamic node names that can be reused, add the following to the cluster configuration:
```

Author: Didn't include ```With this feature, this number has an upper bound roughly equal to the peak number of dynamic nodes working simultaneously.```

Because it is not always true. It it the best case.

```yaml
feature_flags:

>   enable_dynamic_node_name_generation: true

Reviewer: If I understand what's going on here correctly, the feature flag name is weird. It sounds like dynamic node names are always generated, but this feature flag switches the algorithm so it doesn't rely on hostnames.

```

> The node name consists of a prefix and the sequential number of the node within the tenant. By default, the prefix is `slot-`. To override the prefix, you need to add the following to the configuration:

Reviewer: ```suggestion
A generated node name consists of a prefix and a node's sequential number within its tenant. By default, the prefix is `slot-`. To override the prefix, add the following to the configuration:
```
